docs/design/2022-08-11-normalize-client.md

```diff
@@ -0,0 +1,229 @@
+| ~~txnPrewriteBatchSize~~   (delete) | client-java does not support setting |
+| ~~txnCommitBatchSize~~     (delete) | client-java does not support setting |
+| ~~writeBufferSize~~        (delete) | client-java does not support setting |
+| ~~writeThreadPerTask~~     (delete) | client-java does not support setting |
+| ~~retryCommitSecondaryKey~~(delete) | client-java does not support setting |
+| ~~prewriteMaxRetryTimes~~  (delete) | client-java does not support setting |
+
+### ClientSession
+
+The code of the core module takes the `TiSession` class of the client module as the entry, calls
+the `createSnapshot`, `getOrCreateSnapShotCatalog`and `getPdclient` methods, and uses the snapshot,
+catalog, and `pdclient` classes as the cut points to connect other codes of the entire client
+module. So the key is to replace the `tiSession` class and rewrite these methods, so that the core
+module can not only enjoy the upstream functions, but also flexibly rewrite and expand the upstream.
+Therefore, the `ClientSession` class was added, the `TiSession`, `PDClient` and other classes were
+deleted for they will be maintained by the upstream, the Catalog and Snapshot classes were retained, and the `ClientSession` was used as a new
+entry for the core module. `ClientSession` holds the upstream `TiSession` and the current original
+Catalog and Snapshot.
+This code replaces the original call to `TiSession` and creates and calls it via the `ClientSession`
+configuration file.
+
+#### Previous module or method call relationship
+
+```mermaid
+flowchart 
+A[core]
+B[other module]
+C[tiSession]
+D[Snapshot]
+E[Catalog]
+F[PDClient]
+G[....]
+H[TiConfiguaration]
+subgraph <font color='green'>core</font>
+A
+end
+subgraph <font color='green'>other</font>
+B
+end
+subgraph <font color='green'>tikv-client</font>
+C-->D
+C-->E
+C-->F
+C-->H
+C-->G
+end
+A-->C
+B-->C
+
+```
+
+#### Current module or method call relationship
+
+```mermaid
+
+flowchart TD
+A[core]
+B[other module]
+subgraph <font color='green'>core</font>
+A
+end
+subgraph <font color='green'>other</font>
+B
+end
+A-->C
+B-->C
+C[ClientSession]
+D[Snapshot]
+E[Catalog]
+F[TiSession]
+L[PDClient]
+G[...]
+H[tispark/TIConfiguration]
+I[tikv/TIConfiguration]
+C-->|<font color='red'>ticonfiguration converent</font> |F
+subgraph <font color='green'>tispark/tikv-client</font>
+C-->D
+C-->E
+C-->H
+end
+subgraph  <font color='green'>tikv/client-java</font>
+F-->L
+F-->I
+F-->G
+end
+```
+
+## Compatibility
+
+- Not forward compatible, but users will not directly use `tispark client` related modules, so it
+  does not affect users' use.
+- When using `TiSession` afterwards, you need to use `ClientSession` instead. The method provided
+  by `ClientSession` is preferred, and if it is not provided, `getTiKVSession()` is required to
+  indirectly call the upstream richer method.
+- It is recommended to add the `org.tikv.shade` prefix when you need to use the three
```

Comment:
it is recommended or required? if not added prefix will it cause errors? besides, you introduce how to do, it is better to introduce why in design

Reply:
This is the recommended practice. It may be compatible if you don't do this. No guarantees are given that it will work